mat64/dense.go

```diff
@@ -398,29 +400,57 @@ func (m *Dense) Copy(a Matrix) (r, c int) {
 	case RawMatrixer:
 		amat := aU.RawMatrix()
 		if trans {
+			if m.mat.Stride != 1 {
+				m.checkOverlap(amat)
+			}
 			for i := 0; i < r; i++ {
 				blas64.Copy(c,
 					blas64.Vector{Inc: amat.Stride, Data: amat.Data[i : i+(c-1)*amat.Stride+1]},
 					blas64.Vector{Inc: 1, Data: m.mat.Data[i*m.mat.Stride : i*m.mat.Stride+c]})
 			}
 		} else {
-			for i := 0; i < r; i++ {
-				copy(m.mat.Data[i*m.mat.Stride:i*m.mat.Stride+c], amat.Data[i*amat.Stride:i*amat.Stride+c])
+			switch o := offset(m.mat.Data, amat.Data); {
+			case o < 0:
```

Comment:
Does this access ordering actually matter? I know it does for o == 0, but otherwise?